File: testbot/src/main/java/com/annimon/testbot/TestBot.java
package com.annimon.testbot;

import com.annimon.tgbotsmodule.BotHandler;
import com.annimon.tgbotsmodule.BotModule;
import com.annimon.tgbotsmodule.Runner;
import com.annimon.tgbotsmodule.beans.Config;
import com.annimon.tgbotsmodule.services.YamlConfigLoaderService;
import java.util.List;
import org.jetbrains.annotations.NotNull;

public class TestBot implements BotModule {

    public static void main(String[] args) {
        final var profile = (args.length >= 1 && !args[0].isEmpty()) ? args[0] : "";
        Runner.run(profile, List.of(new TestBot()));
    }

    @Override
    public @NotNull BotHandler botHandler(Config config) {
        final var configLoader = new YamlConfigLoaderService();
        final var configFile = configLoader.configFile("testbot", config.getProfile());
        final var botConfig = configLoader.loadFile(configFile, BotConfig.class);
        return new TestBotHandler(botConfig);
    }
}

File: testbot-kotlin/src/main/kotlin/com/annimon/testbotkt/TestBot.kt
package com.annimon.testbotkt

import com.annimon.tgbotsmodule.BotHandler
import com.annimon.tgbotsmodule.BotModule
import com.annimon.tgbotsmodule.BotModuleOptions
import com.annimon.tgbotsmodule.Runner
import com.annimon.tgbotsmodule.beans.Config
import com.annimon.tgbotsmodule.services.YamlConfigLoaderService
import com.fasterxml.jackson.annotation.JsonProperty
import tools.jackson.module.kotlin.KotlinModule

class TestBot : BotModule {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val profile = args.firstOrNull() ?: ""
            Runner.run(profile, listOf(TestBot()))
        }
    }

    override fun botHandler(config: Config): BotHandler {
        val configLoader = YamlConfigLoaderService()
        val configFile = configLoader.configFile("testbot", config.profile)
        val botConfig = configLoader.loadFile(configFile, BotConfig::class.java) {
            it.addModule(KotlinModule.Builder().build())
        }
        val botModuleOptions = BotModuleOptions.createDefault(botConfig.token);
        return TestBotHandler(botModuleOptions, botConfig)
    }
}

data class BotConfig(
        @JsonProperty(required = true)
        val token: String,

        @JsonProperty(required = true)
        val username: String,

        @JsonProperty(required = true)
        val creatorId: Long
)